database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/Post.swift

```diff
@@ -24,14 +24,16 @@ struct Post: Identifiable {
   var title: String
   var body: String
   var starCount: Int
+  var starDictionary: [String: Bool]
 
   init(id: String, uid: String, author: String, title: String, body: String) {
     self.id = id
     self.uid = uid
     self.author = author
     self.title = title
     self.body = body
-    starCount = 0
+    self.starCount = 0
+    self.starDictionary = [:]
```

Comment:
Ah, I see now. So `starDictionary` is a dictionary with user IDs that starred the post, right? If so, I think we should encode it in the name to improve readability e.g. `userIDsStarredBy` or similar.

Reply:
nit: An alternative could be maintaining a starred post IDs collection for a user. A downside: it will be more difficult to calculate starCount for a post.